decorate.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class Decorator
+  include Nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super.slice(0, 10)
+  end
```

Comment:
- [ ] Why dont you use the if condition here since this method is to trim the word If it's longer? I suggest that you first check if the word has a maximum of 10 characters and if so then it trim the word. :blush:

Reply:
> * [ ]  Why dont you use the if condition here since this method is to trim the word If it's longer? I suggest that you first check if the word has a maximum of 10 characters and if so then it trim the word. 😊

- [ ] Good you are slicing but the slice should not always be done, in fact, there might be an issue in the case where the string is not up to 10 in number, so I will also suggest you check with if condition. Mind you, it is 0 to 9, not 10, remember it is index based.